stairway/src/main/java/bio/terra/stairway/impl/FlightFilterAccess.java

```diff
@@ -226,17 +200,19 @@ private void makeFlightFilter(StringBuilder sb, String prefix) {
     }
   }
 
-  /**
-   * Generate the filter for the flight input table for Form 3: the case where there is more that
-   * one predicate.
-   */
-  private void makeInputFilter(StringBuilder sb) {
-    String inter = StringUtils.EMPTY;
-
-    for (FlightFilterPredicate predicate : filter.getInputPredicates()) {
-      String sql = makeInputPredicateSql(predicate);
-      sb.append(inter).append(sql);
-      inter = " OR ";
+  // -- boolean expression methods
+  private String makeBooleanExpressionsFilters(FlightBooleanOperationExpression expression) {
+    switch (expression.getOperation()) {
+      case PREDICATE:
+        return makeInputPredicateSql(expression.getBasePredicate());
+      case AND, OR:
+        return "("
+            + expression.getExpressions().stream()
+                .map(this::makeBooleanExpressionsFilters)
+                .collect(Collectors.joining(expression.getOperation().getSql()))
+            + ")";
+      default:
+        throw new FlightFilterException("Unrecognized boolean operation");
     }
```

Comment:
Use a switch expression here to avoid the need to `throw`.
```suggestion
    return switch (expression.getOperation()) {
      case PREDICATE -> makeInputPredicateSql(expression.getBasePredicate());
      case AND, OR -> "("
            + expression.getExpressions().stream()
                .map(this::makeBooleanExpressionsFilters)
                .collect(Collectors.joining(expression.getOperation().getSql()))
            + ")";
    }
```

Reply:
This isn't on java 17 yet so that syntax doesn't work sadly

Reply:
Oh wait, is stairway not on JDK 17 yet?

Reply:
Yeah.  Libraries get converted later since you don't want to break dependencies and I don't think that all services that use it are on 17 yet

Reply:
Actually stairway _is_ on JDK 17 as of https://github.com/DataBiosphere/stairway/pull/98

Reply:
Oh weird...it was giving me sass